src/Native/Unix/System.Native/pal_io.c

```diff
@@ -329,7 +330,8 @@ int32_t SystemNative_GetReadDirRBufferSize(void)
 #if HAVE_READDIR_R
     // dirent should be under 2k in size
     assert(sizeof(struct dirent) < 2048);
-    return sizeof(struct dirent);
+    // add some extra space so we can align the buffer to dirent.
+    return sizeof(struct dirent) + alignof(struct dirent) - 1;
```

Comment:
@luhenry has started moving System.Native from C++ to C.  Is this going to make it harder?  I'd thought alignof was introduced in C++11, but maybe I'm misremembering.

Reply:
It is also C11. It's that ok? We can replace the alignof with hard coded 8 if it causes build failures.

Reply:
We rely on C99 for System.Native, so if it's a C11 only feature we run into the risk of not supporting older compilers. Couldn't we align to the size of `struct dirent` instead?

Reply:
That would work too. It does mean allocating and zeroing out twice the size (2x280 = 560 on Linux x64). So I prefer using a fixed alignment of 8. I'll adjust it to 8, if you prefer the sizeof still, let me know.